hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/query/AdjacentEdgesQuery.java

```
@@ -0,0 +1,396 @@
+    }
+
+    @Override
+    public Collection<Condition> conditions() {
+        List<Condition> conds = this.selfConditions();
+        if (super.conditionsSize() > 0) {
+            conds.addAll(super.conditions());
+        }
+        return conds;
+    }
+
+    private List<Condition> selfConditions() {
+        if (this.selfConditions != null) {
+            /*
+             * Return selfConditions cache if it has been collected before
+             * NOTE: it's also to keep the serialized condition value
+             */
+            return this.selfConditions;
+        }
+
+        List<Condition> conds = InsertionOrderUtil.newList();
+
+        conds.add(Condition.eq(HugeKeys.OWNER_VERTEX, this.ownerVertex));
+
+        if (this.direction == Directions.BOTH) {
+            conds.add(Condition.in(HugeKeys.DIRECTION, ImmutableList.of(
+                                                       Directions.OUT,
+                                                       Directions.IN)));
+        } else {
+            conds.add(Condition.eq(HugeKeys.DIRECTION, this.direction));
+        }
+
+        if (this.edgeLabels.length == 1) {
+            conds.add(Condition.eq(HugeKeys.LABEL, this.edgeLabels[0]));
+        } else if (this.edgeLabels.length > 1) {
+            conds.add(Condition.in(HugeKeys.LABEL,
+                                   Arrays.asList(this.edgeLabels)));
+        }
+
+        this.selfConditions = conds;
+        return conds;
+    }
+
+    @Override
+    public <T> T condition(Object key) {
+        T cond = this.selfCondition(key);
+        if (cond != null) {
+            return cond;
+        }
+        return super.condition(key);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> T selfCondition(Object key) {
+        if (key == HugeKeys.OWNER_VERTEX) {
+            return (T) this.ownerVertex;
+        }
+        if (key == HugeKeys.DIRECTION) {
+            return (T) this.direction;
+        }
```

The logic in selfCondition() at line 186 should return an array for consistency when edgeLabels.length > 1, or throw the exception immediately instead of having an unreachable return statement.